packages/nx/src/native/tui/app.rs

```diff
@@ -0,0 +1,714 @@
+use color_eyre::eyre::Result;
+use crossterm::event::{KeyCode, KeyEvent, KeyModifiers, MouseEventKind};
+use napi::bindgen_prelude::External;
+use napi::threadsafe_function::{ErrorStrategy, ThreadsafeFunction};
+use ratatui::layout::{Alignment, Rect};
+use ratatui::style::Modifier;
+use ratatui::style::{Color, Style};
+use ratatui::text::{Line, Span};
+use ratatui::widgets::Paragraph;
+use std::default::Default;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::UnboundedSender;
+use tracing::debug;
+
+use crate::native::pseudo_terminal::pseudo_terminal::{ParserArc, WriterArc};
+use crate::native::tui::tui::Tui;
+
+use super::config::TuiConfig;
+use super::task::{Task, TaskResult};
+use super::utils::is_cache_hit;
+use super::{
+    action::Action,
+    components::{
+        countdown_popup::CountdownPopup,
+        help_popup::HelpPopup,
+        tasks_list::{TaskStatus, TasksList},
+        Component,
+    },
+    tui,
+};
+
+#[derive(Default)]
+pub struct AppState {}
+
+pub struct App {
+    pub components: Vec<Box<dyn Component>>,
+    pub quit_at: Option<std::time::Instant>,
+    pub last_tick_key_events: Vec<KeyEvent>,
```

Comment:
This isn't doing anything right now. Let's take it out. We can reintroduce it if we find the need.